### ISSUE TITLE
Breaking changes should be published with major releases

### ISSUE THREAD

**Issue by onmyflow** [prisma/prisma#3980]
## Problem

The recent 2.9.0 library update has created quite a few issues since apparently the `findOne` function can't be called on non-unique columns anymore. This **breaking change** (at least for TypeScript) was quite unexpected.

## Suggested solution

Please release breaking changes like those with major version releases, e.g.: 3.x

## Alternatives

Show a disclaimer in the CLI or on the website or implement feature flags that default to the old behaviour until a next major version release.

## Additional context

While always dragging features like this out to the next major version may hinder innovation inertia, I think this should be taken into consideration for a ORM which is a **major** building block for applications and should thus be relatively stable in its API.

**Comment by janpio** [comment]
We are not aware of the breaking change you describe here as far as I know. Can you please create a new bug issue where you describe what you are experiencing with some code or description to reproduce it? Thanks.

**Comment by thebiglabasky** [comment]
Hi @onmyflow 
We do our best to anticipate breaking changes and communicate them along with our release notes so that users are aware of what they should do while upgrading.
We have a note in our docs explaining our [versioning approach](https://www.prisma.io/docs/more/releases#how-does-prisma-versioning-work) to that end.
If there has been indeed a breaking change slipping in 2.9.0 without notice, that's surely a mistake on our end, and not intentional.

As @janpio mentioned, could you provide more details about what you could do before and can't be done anymore? I have some assumptions based on your problem statement, but having an example would ensure we can double-check and address this.
Thanks.

**Comment by onmyflow** [comment]
Hi @thebiglabasky @janpio, 
yeah sorry... I should've given a better example. 
I thought that this was a commonly known thing.

So we have a prisma schema wich defines a table like this
```
model some_table {
  id               Int              @id @default(autoincrement())
  user_account     String           @unique
  token            String
  update_time      DateTime         @default(now())
  hs_user_accounts hs_user_accounts @relation(fields: [user_account], references: [id])

  @@index([token], name: "idx_hs_user_account_email_request_token")
}
```
Notice that the token column isn't unique and could have several same entries.

What we now used to do, to fetch a entry in the DB was to use the findOne function where we would then search for the token.
```
prisma.some_table.findOne({
  where: {
    token: "some_token"
  }
})
```

This used to work, yet it actually wasn't all to correct, since there was no indication that it should take the first entry.

Now the 2.9.0 release came along, which improved on type checking and thus made the `token` column unavailable for where clauses in the findOne function. This is the completely correct implementation, that's why I didn't want to open a bug report.

But now our problem was that the code couldn't compile anymore, since `tsc` threw hundreds of errors which we first needed to find out why they even existed. The errors given where sth along the line of:
```
Type Error: Couldn't find property 'token' in 'some_table'
```
Which got us _really_ confused.

Hope this clears up my issue

**Comment by thebiglabasky** [comment]
Thanks for these details!

It looks like the previous behaviour wasn't what we expected, so this may have been an unknown "bug" that got fixed with 2.9.0 as a side effect.

For your use case, you can now use `findFirst` which allows doing basically what your `findOne` was doing so that `findOne` is indeed giving the strictness guarantee.
It hopefully comes down to a search and replace.

I acknowledge the mishap and will check with the team if something changed.

**Comment by onmyflow** [comment]
Great! Yeah seems like we didn't use the right function, it was simple to change anyways. I'll close this issue since there seems to be no real problem.
